components/leetcode-area.tsx

```diff
@@ -33,36 +35,69 @@ const LeetCodeArea = () => {
             if (new RegExp('leetcode.com').test(domain) === false) return '';
             userName = getLastPathname(leetcodeUrl) || '';
         } catch (e) {
-            console.error(e);
             return '';
         }
         return userName;
     }, []);
 
     const getLeetCodeInfo = React.useCallback(async () => {
-        console.log('getLeetCodeInfo: is calling');
-        getLeetCodeProfileInfo(leetcodeUserName, QueryType.userProfileQuery).then((output: any) => {
-            dispatch(setLeetcodeUserInfo({ ...output, username: leetcodeUserName }));
-        });
+        const getLeetcodeApi = LeetCodeApi + leetcodeUserName;
 
-        getLeetCodeProfileInfo(leetcodeUserName, QueryType.LangugaeProblemSolvedQuery).then(
-            (output: any) => {
-                dispatch(setLeetcodeLanguageProblemCount(output?.languageProblemCount));
-            }
-        );
-        getLeetCodeProfileInfo(leetcodeUserName, QueryType.TagProblemsCountQuery).then(
-            (output: any) => {
-                dispatch(setLeetcodeTagProblemCounts(output));
-            }
+        const getDataFromDB: any = await GetData(
+            `/api/platform/${Filter.LEETCODE}?source=${getLeetcodeApi}`
         );
+
+        if (getDataFromDB) {
+            console.log(getDataFromDB);
+            dispatch(
+                setLeetcodeUserInfo({
+                    githubUrl: getDataFromDB?.githubUrl,
```

Comment:
@JRakin please write a bit cleaner code. 
initialize the json outside. then pass it to the `setLeetcodeUserInfo`